packages/desktop-client/src/components/util/GenericInput.jsx

```diff
@@ -33,6 +38,37 @@ export function GenericInput({
   const saved = useSelector(state => state.queries.saved);
   const dateFormat = useDateFormat() || 'MM/dd/yyyy';
 
+  const getNumberInputByFormatType = numberFormatType => {
+    switch (numberFormatType) {
+      case 'currency':
+        return (
+          <AmountInput
+            inputRef={inputRef}
+            value={amountToInteger(value)}
+            onUpdate={v => onChange(integerToAmount(v))}
+          />
+        );
+      case 'percentage':
+        return (
+          <PercentInput
+            inputRef={inputRef}
+            value={value}
```

Comment:
Would `defaultValue` work here? and pass that all the way down to the `Input` component instead of `value`. That way we could probably default it to `defaultValue={100}`.

Reply:
Yeah, that's one option! I implemented this, but the new issue is that if you switch Amount -> Percent -> Amount, the amount that you put in initially is overwritten with "+100.00". The root cause is that the amount and percent inputs use the same underlying field (`value`) in the rule, but maybe we could get away with resetting the AmountInput to zero in the same manner. What do you think? It's a bit of a strange UX since you'd still lose the value you specified originally, but the alternative is a data model update and I know some folks are already using this feature...

Reply:
Pushed the `PercentInput` commit and we can discuss the best approach for `AmountInput`